src/output/file_name.rs

```diff
@@ -414,16 +428,13 @@ impl<'a, 'dir, C: Colours> FileName<'a, 'dir, C> {
         let file_style = style_override.unwrap_or(self.style());
         let mut bits = Vec::new();
 
-        let mut display_hyperlink = false;
-        if self.options.embed_hyperlinks == EmbedHyperlinks::On {
+        if self.options.should_embed_hyperlinks() {
             if let Some(abs_path) = self
                 .file
                 .absolute_path()
                 .and_then(|p| p.as_os_str().to_str())
             {
                 bits.push(ANSIString::from(escape::get_hyperlink_start_tag(abs_path)));
-
-                display_hyperlink = true;
             }
         }
```

Comment:
Absolutely not!! x) If the hyperlink is on, then we use (the badly named) `display_hyperlink` to tell us if a hyperlink is used effectively to close the tag afterwards. It’s possible for hyperlinks to be enabled but not used because the path is not convertible to str. So you should keep `display_hyperlink` (if you find a better name it could nice also).